epoxy-adapter/src/main/java/com/airbnb/epoxy/EpoxyVisibilityTracker.java

```diff
@@ -155,22 +250,126 @@ public void onLayoutChange(
         int left, int top, int right, int bottom,
         int oldLeft, int oldTop, int oldRight, int oldBottom
     ) {
-      processChildren();
+      processChangeEvent("onLayoutChange");
     }
 
     @Override
     public void onScrolled(@NonNull RecyclerView recyclerView, int dx, int dy) {
-      processChildren();
+      processChangeEvent("onScrolled");
     }
 
     @Override
     public void onChildViewAttachedToWindow(View child) {
-      processChild(child);
+      processChild(child, false, "onChildViewAttachedToWindow");
     }
 
     @Override
     public void onChildViewDetachedFromWindow(View child) {
-      processChild(child, true);
+      if (visibleDataChanged) {
```

Comment:
@eboudrant It seems like it would be possible for a multi item data change (eg several items removed) to result in several views being detached - this would only account for the first one though.

to simplify things, what if we always processed all children when a view is detached? i'm a bit concerned about the complexity in trying to optimize this, and that we won't be able to correctly handle all cases.

I'm not too familiar with how/when this is called, and it's interaction with the other callbacks, so I'll leave it up to you.

Reply:
With _**always processed all children**_ I am just afraid of scrolling where views detach/re-attach often. I am pretty confident with the `visibleDataChanged` flag.